### PR TITLE
Support Intermediates Certificates in timestamp response

### PR DESCRIPTION
**Pull request by jamie616**
Include parent certificate if need

### CODE REVIEW

File: timestamp_test.go [hunk 1]
```diff
@@ -461,7 +461,104 @@ func TestCreateResponseWithNoTSACertificate(t *testing.T) {
...
+	}
+
+	if !timestampRes.AddTSACertificate {
+		t.Error("TSA certificate must be included in timestamp response")
+	}
+	if fmt.Sprintf("%X", timestamp.Certificates[0].SubjectKeyId) != fmt.Sprintf("%X", parentCert.SubjectKeyId) {
+		t.Errorf("parent certificate is not match with %X", parentCert.SubjectKeyId)
+	}
+}
+func getTSARSAChildKey() *rsa.PrivateKey {
```
Comment (vanbroup) [inline timestamp_test.go:526]: I would be great if we can use more consistent naming here. The test case it called, TestCreateResponseInclude**TSA**And**Intermediates**Certificate but in this code, it's called child and parent keys.

You can use TSA (for the CA) and TSU (for the signing key) as defined in the RFC or you can use the naming as in a hierarchy [root]>[intermediate]>[end-entity] certificate.

Comment (vanbroup) [inline timestamp_test.go:526]: Should we construct one hierarchy for all test cases?
`Root`->`TSA`->`TSU`

Use these certificates and keys for the following test cases:

1. (existing) > Sign with TSU and do not embed certificates
2. (existing) > Sign with TSU and only embed TSU certificate
3. (new) > Sign with TSU and include certificate chain

**Review by vanbroup — CHANGES_REQUESTED** [review]
Thanks, looks good, can you take a look at these suggestions and update the pull request (no need to create a new one).

**Comment by vanbroup** [comment]
Please rebase so that linting can succeed

File: timestamp_test.go (hunk 1 above)
Comment (jamie616) [reply timestamp_test.go]: Following your suggestion, I have modified the test. Thank you for your reminder

**Pull request closed without merging**